### ISSUE TITLE
V5 `signin()` throws error when using `redirect: false`

### ISSUE THREAD

**Issue by herkulano** [nextauthjs/next-auth#9279]
### Provider type

Email

### Environment

System:
OS: macOS 14.1.1
CPU: (8) arm64 Apple M1 Pro
Memory: 872.53 MB / 16.00 GB
Shell: 5.9 - /bin/zsh
Binaries:
Node: 18.18.2 - ~/.nvm/versions/node/v18.18.2/bin/node
npm: 10.2.4 - ~/.nvm/versions/node/v18.18.2/bin/npm
bun: 1.0.14 - ~/.bun/bin/bun
Watchman: 2023.11.20.00 - /opt/homebrew/bin/watchman
Browsers:
Chrome: 119.0.6045.199
Safari: 17.1

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

In `"next-auth": "5.0.0-beta.4"` the `signIn()` with `redirect: false` throws the following error on the client side:

```
Uncaught (in promise) TypeError: Failed to construct 'URL': Invalid base URL
    at signIn (webpack-internal:///(:3000/en/auth/app-pages-browser)/../../node_modules/next-auth/react.js:172:19)
```

### How to reproduce

Install:
```
"next-auth": "5.0.0-beta.4"
```
Run:
```
const authResponse = await signIn("email", {
      email: data.email,
      redirect: false
})
```

### Expected behavior

Should not throw error

**Comment by datagutt** [comment]
This PR seems to have tried to fix an issue with constructing the error URL:
https://github.com/nextauthjs/next-auth/pull/9056

But ``baseUrl`` seems to return the basePath instead of an base url:
https://github.com/nextauthjs/next-auth/blob/656c0cd81732e796e3053c8c83e7c7716fa94b09/packages/next-auth/src/lib/client.ts#L160-L167

**Comment by tills13** [comment]
PR ^ contains fix for this issue.

**Comment by pulgueta** [comment]
I am getting this same issue but with credentials provider instead of email. Already have latest beta update and still not working

**Comment by Meierschlumpf** [comment]
When the next beta release is planed?

**Comment by henriques4nti4go** [comment]
same issue

**Comment by eliza921** [comment]
Same issue here with credentials provider...

**Comment by aliceclv** [comment]
Same issue 👍

**Comment by RoyBkker** [comment]
Having same issue :-(

**Comment by cassioseffrin** [comment]
Same issue for me too!

**Comment by kitchenbeats** [comment]
Same.

**Comment by lucasdale99** [comment]
Same issue!

**Comment by glenntws** [comment]
Would also like to get a new beta release ASAP for next-auth. Currently waiting for this one, since patching it out locally is quite tedious....

**Comment by pulgueta** [comment]
I managed to solve the issue. Simply export the signIn method from the NextAuth class and use it in a server action. Something like this:

![image](https://github.com/nextauthjs/next-auth/assets/38195996/55327454-d1ea-4d64-a9f0-94b06410f316)
![image](https://github.com/nextauthjs/next-auth/assets/38195996/5c191233-de7b-44b0-afb9-152a667afcdb)

**Comment by RicSala** [comment]
> I managed to solve the issue. Simply export the signIn method from the NextAuth class and use it in a server action. Something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/38195996/293748401-55327454-d1ea-4d64-a9f0-94b06410f316.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMjA4NjksIm5iZiI6MTcwNDIyMDU2OSwicGF0aCI6Ii8zODE5NTk5Ni8yOTM3NDg0MDEtNTUzMjc0NTQtZDFlYS00ZDY0LWE5ZjAtOTRiMDY0MTBmMzE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDE4MzYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5MzVkNTY5MTQyNWM5M2UxODZiYjM5YmM0OWUxZGJiMzIwZGMwNGNmMDljMTU2NjBiNTc4NDRjOGNkMDQ0NDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pBCcNNdfik3Ik9Flwj0bA_7_Yge1keKnIUcchiam-UA) ![image](https://private-user-images.githubusercontent.com/38195996/293748494-5c191233-de7b-44b0-afb9-152a667afcdb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMjA4NjksIm5iZiI6MTcwNDIyMDU2OSwicGF0aCI6Ii8zODE5NTk5Ni8yOTM3NDg0OTQtNWMxOTEyMzMtZGU3Yi00NGIwLWFmYjktMTUyYTY2N2FmY2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDE4MzYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MzEwMGU0MmUyMDBkOTBjNGE3MzYwOGY2YjE5Y2JmMzNiMzgwNGIyYzE4ODZkOTc3NmYxNDAwNmRhMmQzODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hbA59SPZs6DZd4Ox9tX56jCf_fBc4sc6X1e0ihr3u5Q)

That's not solving the issue but changing the environment (client --> server)...

**Comment by RicSala** [comment]
Same issue

**Comment by tills13** [comment]
I'll unsubscribe myself from this thread but can we _please_ just thumbs-up the issue or PR instead of commenting "same"

**Comment by beaniebag** [comment]
> > I managed to solve the issue. Simply export the signIn method from the NextAuth class and use it in a server action. Something like this:
> > ![image](https://private-user-images.githubusercontent.com/38195996/293748401-55327454-d1ea-4d64-a9f0-94b06410f316.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMjA4NjksIm5iZiI6MTcwNDIyMDU2OSwicGF0aCI6Ii8zODE5NTk5Ni8yOTM3NDg0MDEtNTUzMjc0NTQtZDFlYS00ZDY0LWE5ZjAtOTRiMDY0MTBmMzE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDE4MzYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5MzVkNTY5MTQyNWM5M2UxODZiYjM5YmM0OWUxZGJiMzIwZGMwNGNmMDljMTU2NjBiNTc4NDRjOGNkMDQ0NDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pBCcNNdfik3Ik9Flwj0bA_7_Yge1keKnIUcchiam-UA) ![image](https://private-user-images.githubusercontent.com/38195996/293748494-5c191233-de7b-44b0-afb9-152a667afcdb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMjA4NjksIm5iZiI6MTcwNDIyMDU2OSwicGF0aCI6Ii8zODE5NTk5Ni8yOTM3NDg0OTQtNWMxOTEyMzMtZGU3Yi00NGIwLWFmYjktMTUyYTY2N2FmY2RiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAyVDE4MzYwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MzEwMGU0MmUyMDBkOTBjNGE3MzYwOGY2YjE5Y2JmMzNiMzgwNGIyYzE4ODZkOTc3NmYxNDAwNmRhMmQzODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hbA59SPZs6DZd4Ox9tX56jCf_fBc4sc6X1e0ihr3u5Q)
> 
> That's not solving the issue but changing the environment (client --> server)...

From authjs.dev's docs.  It seems handling the formdata server side is the given solution:
https://authjs.dev/reference/nextjs#signin

Hopefully a fix for client side to properly handle errors will come later

**Comment by emimaricic** [comment]
same issue... did you guys manage to find a solution?

**Comment by empflow** [comment]
I fixed this by upgrading to 5.0.0-beta.5

**Comment by alexdsgmoura** [comment]
Same issue here with credentials provider.
